api/tests/test_department_api.py

```diff
@@ -0,0 +1,95 @@
+from unittest.mock import patch
+from rest_framework.test import APIClient
+from rest_framework.reverse import reverse
+
+from core.models import User, Department
+
+from api.tests import APIBaseTestCase
+
+client = APIClient()
+
+
+class DepartmentAPITest(APIBaseTestCase):
+    """ Tests for the Department endpoint"""
+
+    def setUp(self):
+        super(DepartmentAPITest, self).setUp()
+        self.admin = User.objects.create_superuser(
+            email='testuser@gmail.com', cohort=19,
+            slack_handle='tester', password='qwerty123'
+        )
+
+        self.department = Department.objects.create(
+            name="Success"
+        )
+        self.department_url = reverse('departments-list')
+        self.token_user = 'testToken'
+
+    def test_non_authenticated_user_get_departments(self):
+        response = client.get(self.department_url)
+        self.assertEqual(response.data, {
+            'detail': 'Authentication credentials were not provided.'
+        })
+
+    @patch('api.authentication.auth.verify_id_token')
+    def test_can_post_department(self, mock_verify_token):
+        mock_verify_token.return_value = {'email': self.admin.email}
+        data = {
+            "name": "People",
+        }
+        response = client.post(
+            self.department_url,
+            data=data,
+            HTTP_AUTHORIZATION="Token {}".format(self.token_user))
+        self.assertIn("name", response.data.keys())
+        self.assertEqual(response.status_code, 201)
+
+    @patch('api.authentication.auth.verify_id_token')
+    def test_cant_post_department_with_same_name(self, mock_verify_token):
+        mock_verify_token.return_value = {'email': self.admin.email}
+        data = {
+            "name": "Success",
+        }
+        response = client.post(
+            self.department_url,
+            data=data,
+            HTTP_AUTHORIZATION="Token {}".format(self.token_user))
+        self.assertIn("name", response.data.keys())
+        self.assertEqual(response.status_code, 400)
+
+    @patch('api.authentication.auth.verify_id_token')
+    def test_editing_department(self,
+                                mock_verify_id_token):
+        mock_verify_id_token.return_value = {'email': self.admin.email}
+        name = "People"
+        data = {"name": name}
+        client.post(
+            self.department_url,
+            data=data,
+            HTTP_AUTHORIZATION="Token {}".format(self.token_user))
+        department_id = Department.objects.get(name=name).id
```

Comment:
Hey @collinmutembei  for this test, we manually queried the `department id` since the `auto increment` id field in the test database does not reset which we don't yet understand why since on other tests it gets reset and we can be sure of the id number, is this the right approach or is there a way to ensure that the field gets reset automatically??

Reply:
@esirK The goal here is to test that the edit functionality provided through PUT works as expected:
- Given an existing department, updates on the name should be reflected
- Only authenticated admin users can edit a department's name

That being said; it's not clear to me why the `id` needs to reset. Find me, so we can discuss this.